### PR TITLE
Jd/attempt sentry cleanup

### PR DESCRIPTION
**Pull request by jnsdls**

### CODE REVIEW

File: sentry.client.config.js [hunk 1]
```diff
@@ -19,9 +32,12 @@ Sentry.init({
...
   replaysSessionSampleRate: 0,
   // record replays for errors
-  replaysOnErrorSampleRate: 1.0,
+  replaysOnErrorSampleRate: 0.1,
   integrations: [new Sentry.Replay()],
 
   ignoreErrors: sentryOptions.ignoreErrors,
   denyUrls: sentryOptions.denyUrls,
+  // only allow thirdweb.com domains
+  allowUrls: ["https://thirdweb.com"],
```
Comment (jnsdls) [inline sentry.client.config.js:41]: this may be too strict

File: sentry.client.config.js [hunk 2]
```diff
@@ -19,9 +32,12 @@ Sentry.init({
   // don't record replays for generic sessions
   replaysSessionSampleRate: 0,
   // record replays for errors
-  replaysOnErrorSampleRate: 1.0,
+  replaysOnErrorSampleRate: 0.1,
```
Comment (jnsdls) [inline sentry.client.config.js:35]: 10% of errors -> replay instead of 100%

File: sentry.client.config.js [hunk 3]
```diff
@@ -3,9 +3,22 @@ import { sentryOptions } from "./sentry.config";
...
+// Only initialize Sentry if the browser is supported
+const isBrowserSupported = matchesUA(navigator.userAgent, {
+  ignoreMinor: true,
+  ignorePatch: true,
+  browsers: packageJson.browserList,
+});
+
+if (!isBrowserSupported) {
+  console.warn("Browser not supported. Sentry will not be initialized.");
+}
```
Comment (jnsdls) [inline sentry.client.config.js:20]: if browser is not supported then don't even enable sentry at all

File: package.json [hunk 4]
```diff
@@ -141,5 +142,9 @@
       "tw-components"
     ],
     "entrypoints": []
-  }
+  },
+  "browserList": [
+    "defaults",
+    "unreleased versions"
+  ]
```
Comment (jnsdls) [inline package.json:149]: recommended options by browserlist

File: next.config.js [hunk 5]
```diff
@@ -107,7 +107,7 @@ module.exports = withPlausibleProxy({
     wSentry(
       moduleExports,
       { silent: true, debug: false },
-      { hideSourceMaps: true, widenClientFileUpload: true },
+      { hideSourceMaps: false, widenClientFileUpload: true },
```
Comment (jnsdls) [inline next.config.js:110]: might help with sourcemap missing?

File: sentry.client.config.js (hunk 1 above)
Comment (MananTank) [reply sentry.client.config.js:41]: Don't we have to use a regex here? or just "thirdweb.com"

Comment (assimovt) [reply sentry.client.config.js:41]: think we should allow using regex to support preview envs  + staging too. lets allow anything ending with thirdweb.com

**Review by assimovt — APPROVED** [review]
just one minor thing, otherwise lets :shipit:

File: sentry.client.config.js (hunk 1 above)
Comment (MananTank) [reply sentry.client.config.js:41]: I don't think we need to report errors from staging - but I think `https://thirdweb.com` will just report errors on home page and not other pages

File: sentry.client.config.js [hunk 6]
```diff
@@ -19,9 +32,12 @@ Sentry.init({
...
   replaysSessionSampleRate: 0,
   // record replays for errors
-  replaysOnErrorSampleRate: 1.0,
+  replaysOnErrorSampleRate: 0.1,
   integrations: [new Sentry.Replay()],
 
   ignoreErrors: sentryOptions.ignoreErrors,
   denyUrls: sentryOptions.denyUrls,
+  // only allow thirdweb.com and thirdweb-preview.com domains
+  allowUrls: [/^(https:\/\/)((thirdweb\.com)|(.*\.thirdweb-preview\.com))$/i],
```
Comment (jnsdls) [inline sentry.client.config.js:41]: @MananTank @assimovt can you guys re-check this regex?

should allow:
- https://thirdweb.com (production)
- https://*.thirdweb-preview.com (preview deployments)

Comment (assimovt) [reply sentry.client.config.js:41]: regex looks good, just thinking if url needs to end $, what if its thirdweb.com/some/page

Comment (MananTank) [reply sentry.client.config.js:41]: This regex only allows the home page ( and only without trailing /  )

image

Comment (MananTank) [reply sentry.client.config.js:41]: This allows this, but shouldn't :

image

Comment (MananTank) [reply sentry.client.config.js:41]: This allows this, but shouldn't :

image

Comment (jnsdls) [reply sentry.client.config.js:41]: @MananTank can you check if they go by domain or if they check the full URL?

**Pull request merged**